src/phy/mod.rs

```diff
@@ -135,12 +135,57 @@ pub use self::tap_interface::TapInterface;
 /// A tracer device for Ethernet frames.
 pub type EthernetTracer<T> = Tracer<T, super::wire::EthernetFrame<&'static [u8]>>;
 
-/// A description of device limitations.
+/// The checksum configuration for a device
+#[derive(Debug, Clone, Copy)]
+pub enum Checksum {
+    /// Validate checksum when receiving and supply checksum when sending
+    Both,
+    /// Validate checksum when receiving
+    Rx,
+    /// Supply checksum before sending
+    Tx,
+    /// Ignore checksum
+    None,
+}
+
+impl Default for Checksum {
+    fn default() -> Checksum {
+        Checksum::Both
+    }
+}
+
+impl Checksum {
+    pub(crate) fn rx(&self) -> bool {
+        match *self {
+            Checksum::Both | Checksum::Rx => true,
+            _ => false
+        }
+    }
+
+    pub(crate) fn tx(&self) -> bool {
+        match *self {
+            Checksum::Both | Checksum::Tx => true,
+            _ => false
+        }
+    }
+}
+
+/// Configuration of checksum capabilities for each applicable protocol
+#[derive(Debug, Clone, Default)]
+pub struct ChecksumCaps {
+    pub ipv4: Checksum,
+    pub udp: Checksum,
```

Comment:
I can imagine that a network card could handle ipv4 checksum but not ipv6, so perhaps split these off into udpv4 and tcpv4?